Orion benchmark visualization

notoraptor opened the pull request:
Hi @bouthilx ! This is a PR for Orion Benchmark visualization. We can already search a benchmark, select algos and tasks, and load graphs. To do:

- Update CI with relevant database (benchmark_db instead of db.pkl)
- Add relevant testing
- Improve interface and fix small bugs

bouthilx commented on LICENSE:
```diff
@@ -0,0 +1,31 @@
+Software License Agreement (BSD License)
+
+ Copyright (c) 2020-2021, Epistímio.
```
```suggestion
 Copyright (c) 2020-2022, Epistímio.
```

bouthilx commented:
It would be great to be able to select/unselect algorithms and tasks from within React instead of making other calls to the backend.

notoraptor commented:
To do:
- fix plot resizing bug
- display task plots horizontally

notoraptor commented:
Hi @bouthilx ! PR updated and ready for review and possibly merging !

- Plots components (assessments, tasks, algorithms) can be (de)selected, and each plot are downloaded once, then reloaded from cached downloaded data at each (de)selection.
- Plots are entirely reloaded (no more display bug)
- Plots are correctly displayed horizontally (no more display bug)
- I added one big long test + CI configuration using `benchmark_db.pkl` database and Orion code from my own repository patching yours (I added a missing import `time`). You can see differences with your branch here: https://github.com/bouthilx/orion/compare/feature/benchmark_webapi...notoraptor:feature/benchmark_webapi?expand=1

Sorry, I went through a bunch or issues to trigger Github CI here (close/reopen this PR, try on new other PRs, push a small commit to `main` to add basic Github worlflow ...), but it finally launched here, and all tests passed.

bouthilx commented on src/__tests__/benchmark.test.js:
```diff
@@ -0,0 +1,180 @@
+  );
+}
+
+test('Test select benchmark', async () => {
```
It would be good to add tests for selections of assessments, tasks and algorithms as well.

bouthilx commented:
Thanks for the PR! Beside some tests that are missing it's all good! :)

notoraptor commented:
> Thanks for the PR! Beside some tests that are missing it's all good! :)

Done!

bouthilx commented on src/__tests__/benchmark.test.js:
```diff
@@ -178,3 +206,480 @@ test('Test select benchmark', async () => {
   );
 });
+
+test('Test (de)select assessments, tasks and algorithms', async () => {
```
Do you think it would be better to separate each section in a different tests to each debugging it? Also I think it would be best to verify that the rest is still displayed when we deselect one item. For instance if the benchmark has AverageRanking and AverageResult, if we deselect AverageRanking, we test that AverageRanking is not displayed anymore, but we should also verify that AverageResult is still displayed.

notoraptor commented:
@bouthilx done!

bouthilx reviewed (approved):
All good! Thanks!
